web/models/course/course_helper.ex

```diff
@@ -0,0 +1,31 @@
+defmodule CoursePlanner.CourseHelper do
+  @moduledoc """
+  This module provides custom functionality for controller over the model
+  """
+  use CoursePlanner.Web, :model
+  import Ecto.DateTime, only: [utc: 0]
+
+  alias CoursePlanner.Repo
+  alias CoursePlanner.Course
+
+  def delete_handler(course) do
+    case course.status do
+      "Planned" -> hard_delete_course(course)
+      _         -> soft_delete_course(course)
+    end
+  end
+
+  defp soft_delete_course(course) do
+    changeset = change(course, %{status: "Deleted", deleted_at: utc()})
```

Comment:
I gave some thought to this `Deleted` status and I think we should remove it, the status is something meaningful to the user and we are overwriting it when we delete the record, so if we happen to need to recover a record we will lose this information.

What do you think of using only the `deleted_at` timestamp as part of soft-deleting records?